client/state/data-layer/wpcom/sites/automated-transfer/status/index.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { delay } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { AUTOMATED_TRANSFER_STATUS_REQUEST } from 'state/action-types';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import {
+	getAutomatedTransferStatus,
+	setAutomatedTransferStatus,
+} from 'state/automated-transfer/actions';
+
+export const requestStatus = ( { dispatch }, action ) => {
+	const { siteId } = action;
+
+	dispatch( http( {
+		method: 'GET',
+		path: `/sites/${ siteId }/automated-transfers/status`,
+		apiVersion: '1',
+	}, action ) );
+};
+
+export const receiveStatus = ( { dispatch }, { siteId }, next, { status, uploaded_plugin_slug } ) => {
+	dispatch( setAutomatedTransferStatus( siteId, status, uploaded_plugin_slug ) );
+	if ( status !== 'complete' ) {
+		delay( dispatch, 3000, getAutomatedTransferStatus( siteId ) );
```

Comment:
The [polling query component](https://github.com/Automattic/wp-calypso/blob/master/client/components/data/query-rewind-restore-status/index.js) is a better pattern for this, because is resilient to connection interruptions, page refreshes and so on. However, this simple data-layer delay allows us to get things working with [hardly any UI changes](https://github.com/Automattic/wp-calypso/pull/17529/files#diff-473121d6da4be762010ef3c8b5df7391R61), so is a good first iteration.